maskit/_masked_circuits.py

```diff
@@ -92,40 +123,53 @@ def differentiable_parameters(self, value) -> None:
         Provides a setter for the differentiable parameters. It is ensured that the
         updated values are written into the underlying :py:attr:`~.parameters`.
         """
+        mask = self._accumulated_mask()
         if self._dynamic_parameters:
-            self.parameters[~self.mask] = value
+            self.parameters[~mask] = value
         else:
-            self.parameters[~self.mask] = value[~self.mask]
+            self.parameters[~mask] = value[~mask]
 
-    @property
-    def mask(self) -> np.ndarray:
+    def full_mask(self, mask_type: Type[Mask]) -> np.ndarray:
         """
-        Accumulated mask of layer, wire, and parameter masks.
+        Accumulated mask of layer, wire, and parameter masks for a given type of Mask.
         Note that this mask is readonly.
         """
-        if Axis.PARAMETERS in self.masks:
-            mask = self.masks[Axis.PARAMETERS].mask.copy()
+        if Axis.PARAMETERS in self.masks and mask_type in self.masks[Axis.PARAMETERS]:
+            result = self.masks[Axis.PARAMETERS][mask_type].mask.copy()
         else:
```

Comment:
Since the two paths do different things but the first *has* to implicitly check the condition, I would just try and except here:
```suggestion
        try:
            result = self.masks[Axis.PARAMETERS][mask_type].mask.copy()
        except KeyError:
```
```